Add a few convenience implementations for UpdateDataValues on IInstance interface

ivarne opened the pull request:
Updating `DataValues` on an instance requires rather ugly code in `DataProcessWrite`

```c#
   var instanceOwnerPartyId = int.Parse(instance.Id.Split("/")[0]);
   var instanceGuid = Guid.Parse(instance.Id.Split("/")[1]);
   var dataValues = new Dictionary<string, string?>{{key, value}};
   return await _instanceClient.UpdateDataValues(instanceOwnerPartyId, instanceGuid, new DataValues{Values = dataValues});
```

This PR is intended to simplify the code required, so that it can be written

```c#
    return await _instanceClient.UpdateDataValue(instance, "key", value);
```
Or if you want to update multiple values
```c#
    var dataValues = new Dictionary<string, string?>()
    {
        {"key1", value1},
        {"key2", value2},
    }
    return await _instanceClient.UpdateDataValues(instance, dataValues);
```

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

ivarne committed:
Add a few convenience implementations for UpdateDataValues on IInstance interface

RonnyB71 commented on src/Altinn.App.Core/Interface/IInstance.cs:
```diff
@@ -92,6 +92,37 @@ public interface IInstance
+        /// <returns>Returns the updated instance.</returns>
+        async Task<Instance> UpdateDataValues(Instance instance, Dictionary<string, string?> dataValues)
+        {
+            var instanceOwnerPartyId = int.Parse(instance.Id.Split("/")[0]);
```
Use InstanceIdetifier to hide the splitting logic.

```csharp
            var instanceIdentifier = new InstanceIdentifier(instance.Id);            
            return await UpdateDataValues(instanceIdentifier.InstanceOwnerPartyId, instanceIdentifier.InstanceGuid, new DataValues{Values = dataValues});
```

RonnyB71 commented:
@ivarne - starting the new year by reviewing open issues - is this still a change you would like to have? I only had a small comment, but are happy to merge after that.

ivarne committed:
Use InstanceIdentifier instead of manually splitting instance.Id

Added a new convenicence constructor to improve type safety when
constructing from an instance

ivarne commented:
Good start of the year ❤️ 
I updated the code, but instead of `new InstanceIdentifier(instance.Id)` I added a new constructor so I could send the instance directly.
